webhook/resourcesemantics/validation/validation.go

```diff
@@ -46,11 +48,15 @@ import (
 
 var errMissingNewObject = errors.New("the new object may not be nil")
 
+// Callback is a generic function to be called by a consumer of validation
+type Callback func(context.Context, unstructured.Unstructured) error
+
 // reconciler implements the AdmissionController for resources
 type reconciler struct {
-	name     string
-	path     string
-	handlers map[schema.GroupVersionKind]resourcesemantics.GenericCRD
+	name      string
+	path      string
+	handlers  map[schema.GroupVersionKind]resourcesemantics.GenericCRD
+	callbacks map[schema.GroupVersionKind]Callback
```

Comment:
can there be more than one callback per type?

Reply:
I considered making it a list of callbacks, but I suspect the pattern will be to create a callback in Main and convert to the concrete type. Then use one callback to shim out to one or many pieces of logic.